libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -707,19 +709,19 @@ private void toggleHtmlMode() {
         mHtmlModeEnabled = !mHtmlModeEnabled;
 
         mEditorFragmentListener.onTrackableEvent(TrackableEvent.HTML_BUTTON_TAPPED);
-        mEditorFragmentListener.onHtmlModeToggledInToolbar();
 
         // Don't switch to HTML mode if currently uploading media
         if (!mUploadingMediaProgressMax.isEmpty() || isActionInProgress()) {
             ToastUtils.showToast(getActivity(), R.string.alert_action_while_uploading, ToastUtils.Duration.LONG);
             return;
         }
 
+        mEditorFragmentListener.onHtmlModeToggledInToolbar();
```

Comment:
I put this call after the `return` above because in the gifs generated and also the gif of the primary issue on `gutenberg-mobile` you can see where the Snackbar is saying that the switch was made even when it wasn't so once the upload is still in progress this line shouldn't be called hence the parent `EditPostActivity` should not be able to invalidate the menu which triggers the snackbar that shows the `Switchted to X Mode` behavior.

Reply:
👍 sounds good

Reply:
Yes, I ran another test on this and realized I would need to change the logic in the `EditPostActivity` a bit because the  `MenuItem` triggers the snackbar saying the switch has happened before the logic in the `GutenbergEditorFragment` is run and it's determined that a switch can be made based on the upload status or not.

Reply:
I did the change here so everything works as expected now a1f54c7

To test you will realize that if you try to switch to HTML mode during upload it will no longer show the snackbar. It will only show the toast telling you that the upload is in progress.

Reply:
thanks for this!